WooCommerce/Classes/ViewRelated/Coupons/CouponListViewModel.swift

```diff
@@ -119,12 +120,41 @@ final class CouponListViewModel {
         let action = AppSettingsAction.updateFeedbackStatus(type: .couponManagement,
                                                             status: .dismissed) { [weak self] result in
             if let error = result.failure {
-                DDLogError("⛔️ Error update feedback visibility for coupon management: \(error)")
+                DDLogError("⛔️ Error updating feedback visibility for coupon management: \(error)")
             }
             self?.isFeedbackBannerEnabledInAppSettings = false
         }
         storesManager.dispatch(action)
     }
+
+    /// Enable coupons for the store
+    ///
+    func enableCoupons() {
+        state = .loading
+        let action = SettingAction.enableCouponSetting(siteID: siteID) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success:
+                self.syncingCoordinator.synchronizeFirstPage(reason: nil, onCompletion: nil)
+            case .failure(let error):
+                DDLogError("⛔️ Error enabling coupon setting: \(error)")
+                self.loadCouponSetting { [weak self] result in
+                    switch result {
+                    case .success(let isEnabled):
+                        if isEnabled {
+                            self?.syncingCoordinator.synchronizeFirstPage(reason: nil, onCompletion: nil)
```

Comment:
is this code path possible 🤔 the store has coupons disabled before, and they tap to enable coupons but the request fails, the app tries loading the coupons setting again and it succeeds?

Reply:
This _could_ be possible, if someone has enabled the setting on the web right before that. But it's a super edge case that is more likely to not happen, and can be recovered by tapping the button again, so yes, let's simplify that. Updated in 21055f9db574a009dcfc89c5b9b7b142a81d88fe.